### ISSUE TITLE
Missing sounds, including "ching" sound effect when a guest makes a purchase.

### ISSUE THREAD

**Issue by Fred-1044276** [OpenRCT2/OpenRCT2#10914]
**OS:** Windows 10
**Version:** 0.2.4
**Commit/Build:** a2d8078

Even if the park has no rides playing music and the guest count is below 400, the "Ching" sound effect heard with a guest purchase either rarely plays or no longer plays. Although I still make money from guests making purchases in the park, the lack of the "Ching" sound effect can be offputting.

This issue currently does not exist in the v0.2.4 stable release.

- Reproducible in RCT2 (vanilla)? No.
- Specific to multiplayer? No.

**Steps to reproduce:**
1. Start up a new scenario. Cannot be a scenario with no money.

2. Build some rides and stalls and build up your guest count.

3. Select a ride or stall that's currently making money and click the "View of ride/attraction" tab again to hear only the ride/stall (for rides, a station/docking platform with an entrance should be selected).

4. If you don't hear the "ching" sound effect at all, then you have successfully reproduced the bug/issue.

5. I have tested this bug on both densely populated parks (+2,000 guests) and sparsely populated parks (1~400 guests). The bug occurs regardless of whether or not the "Show guest purchases as animation" option is toggled on.

**Dump file**
None here, as the bug does not crash the game.

**Screenshots / Video:**
None here.

**Save game:**
[No ching sound bug.zip](https://github.com/OpenRCT2/OpenRCT2/files/4322410/No.ching.sound.bug.zip)
This zip file contains a custom scenario I made and posted in the Marcel's Hangout Discord server. I also used this to reproduce this bug.

**Comment by ocalhoun6** [comment]
I've been seeing the same thing. But it's not just the 'ching' sound.

I've noticed occasional missing ride sounds like the hiss a vertical drop makes, the engine sounds from go karts, and sometimes chain lift sounds. Also -- worst of all -- ducks no longer quack when I click on them.

It's not 100%, though -- it's kind of intermittent. I _am_ still getting the occasional 'ching', but only rarely.

**Comment by AaronVanGeffen** [comment]
If this has started to happen recently, it could be a regression from #10596, which was merged 8 days ago.

**Comment by AaronVanGeffen** [comment]
88fcba35ec12b038800d05e0dab4280a1d329092 does have the sound effects in question.

**Comment by AaronVanGeffen** [comment]
It took about half an hour of compiling and testing, but here's the bisect:

```
git bisect start
# good: [22768ea9150258e774c1b09bcc920a19bd8c4db0] Merge pull request #10596 from tupaschoal/pr-10070-copy
git bisect good 22768ea9150258e774c1b09bcc920a19bd8c4db0
# good: [88fcba35ec12b038800d05e0dab4280a1d329092] Fix #10662: Improve duck cheat tooltips
git bisect good 88fcba35ec12b038800d05e0dab4280a1d329092
# bad: [fd2e3f7b08a810d5c20466d794326e37c46f4b1e] Add #10637 to changelog
git bisect bad fd2e3f7b08a810d5c20466d794326e37c46f4b1e
# bad: [02c135e4fabdb7d30bf31125e9d84038a3d0c904] Clean up many staff functions
git bisect bad 02c135e4fabdb7d30bf31125e9d84038a3d0c904
# bad: [dcf43015597cf71999ff7c10aa2cf8842eb87b5c] Use constant for invalid direction
git bisect bad dcf43015597cf71999ff7c10aa2cf8842eb87b5c
# bad: [8883f428ba67dc671051025c39251a7f2442ff79] Clean up window vars
git bisect bad 8883f428ba67dc671051025c39251a7f2442ff79
# bad: [2cfac9834250e4e7d42710e0e3af974cef69bf38] Make gSavedView use ScreenCoordsXY
git bisect bad 2cfac9834250e4e7d42710e0e3af974cef69bf38
# bad: [09fad0ef7f15ce074c387d1bd9c27789ff827828] Change rct_viewport to use ScreenCoordsXY
git bisect bad 09fad0ef7f15ce074c387d1bd9c27789ff827828
# good: [87f724d038fa59f8d75ffe318bcd1763899f9fb0] Use Coords objects for viewport_create()
git bisect good 87f724d038fa59f8d75ffe318bcd1763899f9fb0
# first bad commit: [09fad0ef7f15ce074c387d1bd9c27789ff827828] Change rct_viewport to use ScreenCoordsXY
```

In conclusion, this bug is caused by 09fad0ef7f15ce074c387d1bd9c27789ff827828
